### PR TITLE
[Bug](pipeline) Fix bugs to pass TPCDS cases

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.5 seconds
 load time: 664 seconds
 storage size: 17123555171 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221220065953_clickbench_pr_65485.html

File: be/src/pipeline/exec/union_source_operator.cpp [hunk 1]
```diff
@@ -61,6 +61,9 @@ Status UnionSourceOperator::get_block(RuntimeState* state, vectorized::Block* bl
         std::unique_ptr<vectorized::Block> output_block;
         int child_idx = 0;
         _data_queue->get_block_from_queue(&output_block, &child_idx);
+        if (!output_block) {
+            return Status::OK();
+        }
```
Comment (HappenLee) [inline be/src/pipeline/exec/union_source_operator.cpp:66]: do not directly return ？ use lower code to dispose the `SourceState`

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 2]
```diff
@@ -758,7 +758,10 @@ Status HashJoinNode::sink(doris::RuntimeState* state, vectorized::Block* in_bloc
     }
 
     if (_should_build_hash_table && eos) {
-        child(1)->close(state);
+        // For pipeline engine, children should be closed once this pipeline task is finished.
+        if (!state->enable_pipeline_exec()) {
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.cpp:762]: maybe should do same check in 808line, !_should_build_hash_table

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**